### PR TITLE
Make `String` compatible with the markdown filter.

### PR DESCRIPTION
**Pull request by Wilkuu-2**
Hello, I recently decided to switch from Tera to Askama, and I'm enjoying it so far. 

Yesterday, I stumbled upon issue #719 and decided to see if I can fix the issue with the markdown filter, so I don't have to include Comrak in my own crate as a workaround. 
What I found out is that the issue with using `String` is that the filter moves the value out of the struct. This is fine for `&str` considering `&str` is already a reference. 
```rs
    /// Code generated by Template derive macro  
    fn render_into(&self, writer: &mut (impl ::std::fmt::Write + ?Sized)) -> ::askama::Result<()> {
    ...
            expr0 = &::askama::filters::markdown(::askama::Html, self.content, ::core::option::Option::None)?,
            //                                                   ^^^^^^^^^^^^ 
            //                                              Value gets moved out by filter 
    ...
```

Making the generated code borrow the value did not create any problems in tests.

This PR also adds a test for passing `String`s to the markdown filter and simplifies the function
signature for the filter to accept `&str`.

### CODE REVIEW

File: askama_derive/src/generator.rs [hunk 1]
```diff
@@ -1200,7 +1200,7 @@ impl<'a> Generator<'a> {
         };
 
         buf.write(&format!(
-            "::askama::filters::markdown({}, ",
+            "::askama::filters::markdown({}, &",
```
Comment (GuillaumeGomez) [inline askama_derive/src/generator.rs:1203]: Normally, only this change should be needed and not the rest. Also a lot more types are actually implementing `AsRef<str>`, so this is a regression in a sense since now it only works with `String` and `str`.

Comment (djc) [reply askama_derive/src/generator.rs:1203]: Yeah, that's exactly what I'm worried about too.

File: askama/src/filters/mod.rs [hunk 2]
```diff
@@ -342,14 +342,14 @@ pub fn wordcount<T: fmt::Display>(s: T) -> Result<usize> {
 }
 
 #[cfg(feature = "markdown")]
-pub fn markdown<E, S>(
+pub fn markdown<E,S>(
```
Comment (djc) [inline askama/src/filters/mod.rs:345]: You've left a bunch of trivial changes in this PR, please actually revert the changes to the previous state.

File: testing/tests/markdown.rs [hunk 3]
```diff
@@ -70,3 +71,18 @@ before\
...
     );
 }
+
+#[derive(Template)]
+#[template(source = "{{content|markdown}}", ext = "html", print="code")]
+struct MarkdownStringTemplate{
+    pub content: String
+}
+
+#[test]
```
Comment (GuillaumeGomez) [inline testing/tests/markdown.rs:81]: Could add a comment saying exactly what this test is checking please? Much better if this breaks in the future to understand what the new changes did wrong.

Comment (Wilkuu-2) [reply testing/tests/markdown.rs:80]: I actually wanted to ask if this test is necessary. Since it only tests if the markdown filter accepts `String`.

Comment (GuillaumeGomez) [reply testing/tests/markdown.rs:80]: Well, your use-case is valid and I think it's important for us to ensure that it won't regress again. Nothing prevents you to add more checks in this test though. ;)

Comment (Wilkuu-2) [reply testing/tests/markdown.rs:81]: It's good to hear that the test can stay. 

I think this check is enough for this use-case and this PR. I'll consider adding more checks in a future PR though.

Comment (GuillaumeGomez) [reply testing/tests/markdown.rs:81]: Fine by me. 👍

**Comment by GuillaumeGomez** [comment]
Ok, now please run `cargo fmt` and then [squash your commits](https://blog.guillaume-gomez.fr/Git-tips/1/2).

**Comment by djc** [comment]
@GuillaumeGomez if there is a single conceptual change we don't need the submitter to squash -- we can just "Squash and merge" instead.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Ah true. I keep forgetting about it.

**Comment by djc** [comment]
Thanks!

**Comment by Apollo-XIV** [comment]
fixed an issue for me that almost made me swap to a different templating engine before i saw this thread! thank you all :D
